whylog/teacher/rule_validation_problems.py

```diff
@@ -0,0 +1,89 @@
+import itertools
+from collections import namedtuple
+
+ValidationResult = namedtuple('ValidationResult', ['errors', 'warnings'])
+
+
+class ValidationResult(object):
+    def __init__(self, errors, warnings):
+        """
+        :param errors: problems preventing form rule saving
+        :param warnings: problems accepted while rule saving
+        :type errors: [RuleValidationProblem]
+        :type warnings: [RuleValidationProblem]
+        """
+        self.errors = errors
+        self.warnings = warnings
+
+    @classmethod
+    def result_from_results(cls, results):
+        errors = sum([result.errors for result in results], [])
+        warnings = sum([result.warnings for result in results], [])
+        return ValidationResult(errors, warnings)
+
+    def acceptable(self):
+        return not self.errors
+
+    def select_parser_problems(self, line_id):
+        return [
+            problem for problem in itertools.chain(self.warnings, self.errors)
+            if isinstance(problem, ParserValidationProblem) \
+                and problem.get_line_id() == line_id
+        ]  # yapf: disable
+
+    def select_constraint_problems(self, constraint_id):
+        return [
+            problem for problem in itertools.chain(self.warnings, self.errors)
+            if isinstance(problem, ConstraintValidationProblem) \
```

Comment:
`...if problem.concerns_constraint() and ...`

Also, why do you need this method? Andrzej is going to use it? I don't think so.